src/utils/narrow.js

```diff
@@ -304,18 +311,23 @@ export const canSendToNarrow = (narrow: Narrow): boolean =>
     search: () => false,
   });
 
-export const getNarrowFromMessage = (message: Message | Outbox, ownEmail: string) => {
-  if (Array.isArray(message.display_recipient)) {
-    const recipient =
-      message.display_recipient.length > 1
-        ? message.display_recipient.filter(x => x.email !== ownEmail)
-        : message.display_recipient;
-    return groupNarrow(recipient.map(x => x.email));
-  }
-
-  if (message.subject && message.subject.length) {
-    return topicNarrow(message.display_recipient, message.subject);
+/**
+ * Careful: quirky behavior on a stream message with empty topic.
+ *
+ * If you think you want to reuse this function: study carefully; maybe
+ * refactor it to something cleaner first; if not, then definitely document
+ * its quirky behavior.
+ */
+// TODO: do that, or just make this a private local helper of its one caller
```

Comment:
> // TODO: do that, or just make this a private local helper of its one caller

Yeah, this might be the way to go. #4317 contains an attempt to clean it up, but I'm not sure about the value of that particular attempt, especially since there is just one caller, as you mention.